src/adapter/src/catalog.rs

```diff
@@ -3296,30 +3298,78 @@ impl Catalog {
                 }
                 // If we were missing a dependency, wait for it to be added.
                 Err(AdapterError::PlanError(plan::PlanError::InvalidId(missing_dep))) => {
-                    awaiting_dependencies
+                    awaiting_id_dependencies
                         .entry(missing_dep)
                         .or_default()
                         .push((id, name, def));
                     continue;
                 }
+                // If we were missing a dependency, wait for it to be added.
+                Err(AdapterError::PlanError(plan::PlanError::Catalog(
+                    SqlCatalogError::UnknownItem(missing_dep),
+                ))) => {
+                    match GlobalId::from_str(&missing_dep) {
```

Comment:
What's an example dependency name that would trigger the error case here? Dependency names are sometimes global ids and sometimes not?

Reply:
Yeah, the values that gets plumbed into `UnknownItem` is sometimes a [stringified version of a `GloablId`](https://github.com/MaterializeInc/materialize/blob/main/src/adapter/src/coord/sequencer.rs#L3299), and sometimes a [stringified version of the item name](https://github.com/MaterializeInc/materialize/blob/main/src/adapter/src/catalog.rs#L1154). 

I don't think the case of the `GlobalId` is currently on this code path, but included it in case someone adds it in the future based on the examples of `UnknownItem`'s usage elsewhere in the code––if you poke around, you'll see that every instance of `UnknownItem` uses a stringified ID except for the one call to resolve that I linked to.